### PR TITLE
Docker builds for other CUDA versions, improve CI

### PR DESCRIPTION
**Pull request by epwalsh**
Feature request from our conversation here: https://github.com/allenai/allennlp/pull/4793#discussion_r523289015. This makes it so we'll build Docker images for both CUDA 10.2 and CUDA 11.0. These will be published as `allennlp/allennlp:v1.2.1-cuda10.2` and `allennlp/allennlp:v1.2.1-cuda11.0`, respectively. We will still also provide a `latest` tag which for now will default to the CUDA 10.2 image for consistency with PyTorch.

This also merges our two GH Actions workflows (`.github/workflows/master.yml` and `.github/workflows/pull_request.yml`) into a single one: `.github/workflows/ci.yml`.

I still have a couple things to do:
- [x] Build our official images against different CUDA/torch versions, and push them to Docker hub with a tag that identifies the CUDA/torch version.
- [x] Test these images in our pull request CI. Maybe it's time to combine the `pull_request.yml` and `master.yml` workflows. Could just call it `ci.yml`.
- [ ] Make similar changes to our image builds in allennlp-models.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -247,45 +274,6 @@ with the `allennlp` command (whether you installed from `pip` or from source).
 `allennlp` has various subcommands such as `train`, `evaluate`, and `predict`.
 To see the full usage information, run `allennlp --help`.
 
-## Docker images
-
-AllenNLP releases Docker images to [Docker Hub](https://hub.docker.com/r/allennlp/) for each release.  For information on how to run these releases, see [Installing using Docker](#installing-using-docker).
-
-### Building a Docker image
```
Comment (dirkgr) [inline README.md:254]: Why did this go away?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -263,71 +272,82 @@ jobs:
...
+      if: github.event_name == 'push' && matrix.cuda == '10.2'
       run: |
         docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
         docker push $DOCKER_IMAGE_NAME
 
     - name: Upload release image
-      # Only run this for releases on the main repo, not forks.
-      if: github.repository == 'allenai/allennlp' && github.event_name == 'release'
+      # Only run this for releases.
+      if: github.event_name == 'release'
```
Comment (dirkgr) [inline .github/workflows/ci.yml:326]: Why did all of these `if` statements change?

File: README.md (hunk 1 above)
Comment (epwalsh) [reply README.md:254]: I just moved it into the "Installing via Docker" section.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (epwalsh) [reply .github/workflows/ci.yml:326]: We checked if `github.repository == 'allenai/allennlp'` above, so it was redundant to have that again.

File: README.md (hunk 1 above)
Comment (dirkgr) [reply README.md:254]: You did 🙈

**Review by dirkgr — APPROVED** [review]

**Pull request merged**